clang/lib/CodeGen/CGDebugInfo.cpp

```diff
@@ -6056,8 +6056,19 @@ bool clang::CodeGen::noSystemDebugInfo(const Decl *D,
     if (isa<TypedefDecl>(D))
       return false;
 
-    // -fno-system-debug was used.  Do not generate debug info.
-    if (CGM.getCodeGenOpts().NoSystemDebug)
+#ifndef WINDOWS
+    // Generate debug for _M_insert_aux to allow debugging for
+    // deque's value_type.
+    const auto *Constructor = dyn_cast<CXXConstructorDecl>(D);
+    if (Constructor && Constructor->getNameAsString() == "deque")
+      return false;
+    const auto *Method = dyn_cast<CXXMethodDecl>(D);
+    if (Method && Method->getNameAsString() == "_M_insert_aux")
+      return false;
+#endif
+
+    // -fsystem-debug was not used.  Do not generate debug info.
```

Comment:
Can be rewritten.

// Generate debug info only if -fsystem-debug is specified

Reply:
Changed.